### ISSUE TITLE
Inconsistent InfFE::Radial::decay_deriv() in 1D, 2D?

### ISSUE THREAD

**Issue by roystgnr** [libMesh/libmesh#2232]
Pinging @BalticPinguin since I don't think any other currently-active developers are familiar with the infinite elements.

Do we have an inconsistency between decay() and decay_deriv() in 2D and 1D?  The decay function in those two cases is listed as a "TODO:[DD]" and returns 0, but the decay_deriv() still returns -.5 (consistent for the 3D case only), not 0.

Or is this just a moot problem, because decay=0 makes the 2D and 1D shape() values unusable?  I know our only InfFE examples were for 3D elements, but I didn't guess that was because our only InfFE *support* was for 3D elements!

**Comment by BalticPinguin** [comment]
you are right; it is in a sense inconsistent. But actually we _do_ support only 3D elements. _e.g._ in combine_base_radial(Elem) (``src/fe/inf_fe.C``, l 771) is a ``libmesh_not_implemented()``-remark.
I think it should be not demanding to add their support, if requested, I can looup the decay in in 2D (in 1D 'decay' is 1)

**Comment by roystgnr** [comment]
Thanks!

I'd actually request *not* adding their support for now, because I'm looking at some refactoring that would create merge conflicts, but if it's not hard then it might be a nice idea in the long term.  Infinite Elements are kind of the redheaded-stepchild of libMesh, since their original author stopped developing them so long ago, and having broader support (especially on super-cheap 1D+2D problems) might make it easier to add more unit tests and reduce the likelihood of inadvertent regressions.
